### ISSUE TITLE
in_exec plugin command is only executed once

### ISSUE THREAD

**Issue by yiip87** [fluent/fluent-bit#5221]
## Bug Report

**Describe the bug**
The `in_exec` plugin only executes the specified command once at startup.

**To Reproduce**
- Run `bin/fluent-bit -vv -i exec -p 'command=echo test' -o stdout`

**Expected behavior**
`fluent-bit` should print a message like the following to stdout every second (default for `interval_sec` is one second):
```
[0] exec.0: [1648767130.968141276, {"exec"=>"test"}]
```

**Your Environment**

* Version used: 1.9.0, 1.9.1 and 1.9.2 / 7a9ee966
* Configuration: `bin/fluent-bit -vv -i exec -p 'command=echo test' -o stdout`
* Environment name and version (e.g. Kubernetes? What version?): N/A
* Server type and version: Baremetal
* Operating System and version: Debian 11 (Bullseye)
* Filters and plugins: `in_exec`

**Additional context**
<details><summary>fluent-bit log output</summary>

```
build:~/fluent-bit/build$ bin/fluent-bit -vv -i exec -p 'command=echo test' -o stdout
Fluent Bit v1.9.2
* Copyright (C) 2015-2021 The Fluent Bit Authors
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2022/03/31 22:51:22] [ info] Configuration:
[2022/03/31 22:51:22] [ info]  flush time     | 1.000000 seconds
[2022/03/31 22:51:22] [ info]  grace          | 5 seconds
[2022/03/31 22:51:22] [ info]  daemon         | 0
[2022/03/31 22:51:22] [ info] ___________
[2022/03/31 22:51:22] [ info]  inputs:
[2022/03/31 22:51:22] [ info]      exec
[2022/03/31 22:51:22] [ info] ___________
[2022/03/31 22:51:22] [ info]  filters:
[2022/03/31 22:51:22] [ info] ___________
[2022/03/31 22:51:22] [ info]  outputs:
[2022/03/31 22:51:22] [ info]      stdout.0
[2022/03/31 22:51:22] [ info] ___________
[2022/03/31 22:51:22] [ info]  collectors:
[2022/03/31 22:51:22] [ info] [fluent bit] version=1.9.2, commit=7a9ee9666b, pid=37537
[2022/03/31 22:51:22] [debug] [engine] coroutine stack size: 24576 bytes (24.0K)
[2022/03/31 22:51:22] [debug] [storage] [cio stream] new stream registered: exec.0
[2022/03/31 22:51:22] [ info] [storage] version=1.1.6, type=memory-only, sync=normal, checksum=disabled, max_chunks_up=128
[2022/03/31 22:51:22] [ info] [cmetrics] version=0.3.0
[2022/03/31 22:51:22] [debug] [exec:exec.0] created event channels: read=21 write=22
[2022/03/31 22:51:22] [debug] [input:exec:exec.0] interval_sec=1 interval_nsec=0 oneshot=0 buf_size=4096
[2022/03/31 22:51:22] [debug] [stdout:stdout.0] created event channels: read=23 write=24
[2022/03/31 22:51:22] [debug] [router] default match rule exec.0:stdout.0
[2022/03/31 22:51:22] [ info] [sp] stream processor started
[2022/03/31 22:51:22] [ info] [output:stdout:stdout.0] worker #0 started
[2022/03/31 22:51:22] [debug] [input chunk] update output instances with new chunk size diff=22
[2022/03/31 22:51:23] [trace] [task 0x7f666400d620] created (id=0)
[2022/03/31 22:51:23] [debug] [task] created task=0x7f666400d620 id=0 OK
[2022/03/31 22:51:23] [debug] [output:stdout:stdout.0] task_id=0 assigned to thread #0
[0] exec.0: [1648767082.439953496, {"exec"=>"test"}]
[2022/03/31 22:51:23] [debug] [out flush] cb_destroy coro_id=0
[2022/03/31 22:51:23] [trace] [coro] destroy coroutine=0x7f665c001040 data=0x7f665c001058
[2022/03/31 22:51:23] [trace] [engine] [task event] task_id=0 out_id=0 return=OK
[2022/03/31 22:51:23] [debug] [task] destroy task=0x7f666400d620 (task_id=0)
^C[2022/03/31 22:51:51] [engine] caught signal (SIGINT)
[2022/03/31 22:51:51] [trace] [engine] flush enqueued data
[2022/03/31 22:51:51] [ warn] [engine] service will shutdown in max 5 seconds
[2022/03/31 22:51:52] [ info] [engine] service has stopped (0 pending tasks)
[2022/03/31 22:51:52] [ info] [output:stdout:stdout.0] thread worker #0 stopping...
[2022/03/31 22:51:52] [ info] [output:stdout:stdout.0] thread worker #0 stopped
```

</details>

**Comment by jonschwartz** [comment]
Also seeing this on a digital ocean droplet running 1.9.1 (upgraded from 1.9.0 where I was also seeing it) and Ubuntu 20.04

**Comment by yiip87** [comment]
Couldn't get 1.8 to build yesterday because it kept failing because of missing luajit files. Built 1.8 without luajit today and can confirm that this bug doesn't exist in 1.8.

Built 1.9 from master with @sparkyb's patch and the issue is gone:
<details><summary>fluent-bit log output</summary>

```
build:~/fluent-bit/build$ bin/fluent-bit -vv -i exec -p 'command=echo test' -o stdout
Fluent Bit v1.9.2
* Copyright (C) 2015-2021 The Fluent Bit Authors
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2022/04/01 20:15:53] [ info] Configuration:
[2022/04/01 20:15:53] [ info]  flush time     | 1.000000 seconds
[2022/04/01 20:15:53] [ info]  grace          | 5 seconds
[2022/04/01 20:15:53] [ info]  daemon         | 0
[2022/04/01 20:15:53] [ info] ___________
[2022/04/01 20:15:53] [ info]  inputs:
[2022/04/01 20:15:53] [ info]      exec
[2022/04/01 20:15:53] [ info] ___________
[2022/04/01 20:15:53] [ info]  filters:
[2022/04/01 20:15:53] [ info] ___________
[2022/04/01 20:15:53] [ info]  outputs:
[2022/04/01 20:15:53] [ info]      stdout.0
[2022/04/01 20:15:53] [ info] ___________
[2022/04/01 20:15:53] [ info]  collectors:
[2022/04/01 20:15:53] [ info] [fluent bit] version=1.9.2, commit=7a9ee9666b, pid=122891
[2022/04/01 20:15:53] [debug] [engine] coroutine stack size: 24576 bytes (24.0K)
[2022/04/01 20:15:53] [debug] [storage] [cio stream] new stream registered: exec.0
[2022/04/01 20:15:53] [ info] [storage] version=1.1.6, type=memory-only, sync=normal, checksum=disabled, max_chunks_up=128
[2022/04/01 20:15:53] [ info] [cmetrics] version=0.3.0
[2022/04/01 20:15:53] [debug] [exec:exec.0] created event channels: read=21 write=22
[2022/04/01 20:15:53] [debug] [input:exec:exec.0] interval_sec=1 interval_nsec=0 oneshot=0 buf_size=4096
[2022/04/01 20:15:53] [debug] [stdout:stdout.0] created event channels: read=23 write=24
[2022/04/01 20:15:53] [debug] [router] default match rule exec.0:stdout.0
[2022/04/01 20:15:53] [ info] [sp] stream processor started
[2022/04/01 20:15:53] [ info] [output:stdout:stdout.0] worker #0 started
[2022/04/01 20:15:54] [debug] [input chunk] update output instances with new chunk size diff=22
[2022/04/01 20:15:55] [trace] [task 0x7ff95000d620] created (id=0)
[2022/04/01 20:15:55] [debug] [task] created task=0x7ff95000d620 id=0 OK
[2022/04/01 20:15:55] [debug] [output:stdout:stdout.0] task_id=0 assigned to thread #0
[0] exec.0: [1648844154.317210513, {"exec"=>"test"}]
[2022/04/01 20:15:55] [debug] [out flush] cb_destroy coro_id=0
[2022/04/01 20:15:55] [trace] [coro] destroy coroutine=0x7ff948001040 data=0x7ff948001058
[2022/04/01 20:15:55] [debug] [input chunk] update output instances with new chunk size diff=22
[2022/04/01 20:15:55] [trace] [engine] [task event] task_id=0 out_id=0 return=OK
[2022/04/01 20:15:55] [debug] [task] destroy task=0x7ff95000d620 (task_id=0)
[2022/04/01 20:15:56] [trace] [task 0x7ff95000d6f0] created (id=0)
[2022/04/01 20:15:56] [debug] [task] created task=0x7ff95000d6f0 id=0 OK
[2022/04/01 20:15:56] [debug] [output:stdout:stdout.0] task_id=0 assigned to thread #0
[0] exec.0: [1648844155.317492710, {"exec"=>"test"}]
[2022/04/01 20:15:56] [debug] [out flush] cb_destroy coro_id=1
[2022/04/01 20:15:56] [trace] [coro] destroy coroutine=0x7ff948007340 data=0x7ff948007358
[2022/04/01 20:15:56] [debug] [input chunk] update output instances with new chunk size diff=22
[2022/04/01 20:15:56] [trace] [engine] [task event] task_id=0 out_id=0 return=OK
[2022/04/01 20:15:56] [debug] [task] destroy task=0x7ff95000d6f0 (task_id=0)
[2022/04/01 20:15:57] [trace] [task 0x7ff95000f9d0] created (id=0)
[2022/04/01 20:15:57] [debug] [task] created task=0x7ff95000f9d0 id=0 OK
[2022/04/01 20:15:57] [debug] [output:stdout:stdout.0] task_id=0 assigned to thread #0
[0] exec.0: [1648844156.317275375, {"exec"=>"test"}]
[2022/04/01 20:15:57] [debug] [out flush] cb_destroy coro_id=2
[2022/04/01 20:15:57] [trace] [coro] destroy coroutine=0x7ff9480073b0 data=0x7ff9480073c8
[2022/04/01 20:15:57] [debug] [input chunk] update output instances with new chunk size diff=22
[2022/04/01 20:15:57] [trace] [engine] [task event] task_id=0 out_id=0 return=OK
[2022/04/01 20:15:57] [debug] [task] destroy task=0x7ff95000f9d0 (task_id=0)
^C[2022/04/01 20:15:57] [engine] caught signal (SIGINT)
[2022/04/01 20:15:57] [trace] [engine] flush enqueued data
[2022/04/01 20:15:57] [trace] [task 0x7ff95000fb50] created (id=0)
[2022/04/01 20:15:57] [debug] [task] created task=0x7ff95000fb50 id=0 OK
[2022/04/01 20:15:57] [debug] [output:stdout:stdout.0] task_id=0 assigned to thread #0
[2022/04/01 20:15:57] [ warn] [engine] service will shutdown in max 5 seconds
[0] exec.0: [1648844157.317610886, {"exec"=>"test"}]
[2022/04/01 20:15:57] [debug] [out flush] cb_destroy coro_id=3
[2022/04/01 20:15:57] [trace] [coro] destroy coroutine=0x7ff948007420 data=0x7ff948007438
[2022/04/01 20:15:57] [trace] [engine] [task event] task_id=0 out_id=0 return=OK
[2022/04/01 20:15:57] [debug] [task] destroy task=0x7ff95000fb50 (task_id=0)
[2022/04/01 20:15:58] [ info] [engine] service has stopped (0 pending tasks)
[2022/04/01 20:15:58] [ info] [output:stdout:stdout.0] thread worker #0 stopping...
[2022/04/01 20:15:58] [ info] [output:stdout:stdout.0] thread worker #0 stopped
```

</details>

**Comment by agup006** [comment]
@pwhelan @nokute78 would you be able to help fix the ConfigMap?

**Comment by nokute78** [comment]
#5227 will fix this issue.

Another config map bug will be fixed by #5127
